File: rust/71.simplify-path.rs
/*
 * @lc app=leetcode id=71 lang=rust
 *
 * [71] Simplify Path
 */

// @lc code=start
impl Solution {
    pub fn simplify_path(path: String) -> String {
        let mut paths = Vec::new();
        for p in path.split('/') {
            match p {
                "" | "." => continue,
                ".." => {
                    paths.pop();
                }
                _ => {
                    paths.push(p);
                }
            };
        }
        format!("/{}", paths.join("/"))
    }
}
// @lc code=end

File: javascript/71.simplify-path.js
/*
 * @lc app=leetcode id=71 lang=javascript
 *
 * [71] Simplify Path
 */

// @lc code=start
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function (path) {
  return (
    "/" +
    path
      .split("/")
      .reduce((acc, p) => {
        if (p === "..") acc.pop();
        else if (p !== "" && p !== ".") acc.push(p);
        return acc;
      }, [])
      .join("/")
  );
};
// @lc code=end
